src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -237,11 +240,31 @@ private DRSObject drsObjectFromFSFile(
       FSFile fsFile, String snapshotId, AuthenticatedUserRequest authUser) {
     DRSObject fileObject = makeCommonDrsObject(fsFile, snapshotId);
 
-    GoogleBucketResource bucketResource =
-        resourceService.lookupBucketMetadata(fsFile.getBucketResourceId());
+    List<DRSAccessMethod> accessMethods;
+    CloudPlatformWrapper platformWrapper = CloudPlatformWrapper.of(fsFile.getCloudPlatform());
+    if (platformWrapper.isGcp()) {
+      accessMethods = getDrsAccessMethodsOnGcp(fsFile, authUser);
+    } else if (platformWrapper.isAzure()) {
+      accessMethods = getDrsAccessMethodsOnAzure(fsFile);
+    } else {
+      throw new IllegalArgumentException("Unrecognized cloud platform");
+    }
```

Comment:
This is another case that could be done inside the platform wrapper code. The method could accept `(ResourceService, FSFile, AuthenticatedUserRequest)` and return `List<DRSAccessMethod>`

Putting code into the wrapper class is a start toward having a service level abstraction for different platforms. Once we have the code isolated in the wrapper, we can create different service classes for each collection of operations.

Reply:
Yeah, that's a fair point.  It still feels a little weird but I'll move it in there to to see...

Reply:
Hmm...this got super gross since these methods have a bunch of dependencies on member variables and shared methods in this class which is one of the issues I have with the cloud wrapper method.  It's fine for the simplest cases but I think it makes things more confusing in a lot of cases like this

Reply:
I'd like to know more about what problems you can into, as the dependencies don't seem problematic to me.

The only dependencies for `getDrsAccessMethodsOnGcp()` are `resourceService`, `getDrsAccessMethodsOnGcp()` and `getDrsAccessMethodsOnAzure()` (which can both be converted to `static` methods.)

The only dependency for `getDrsAccessMethodsOnAzure()` is `resourceService`.